### PR TITLE
implement custom types defined in beefy proto file

### PR DESCRIPTION
**Pull request by oshorefueled**
* add beefy to default allowed light clients

### CODE REVIEW

File: proto/ibc/lightclients/beefy/v1/beefy.proto [hunk 1]
```diff
@@ -140,6 +140,9 @@ message Header {
 
   // optional payload to update the mmr root hash.
   MmrUpdateProof mmr_update_proof = 4 [(gogoproto.nullable) = true];
+
+  // previously finalized block of the relay chain
+  uint64 previously_finalized = 5;
```
Comment (seunlanlege) [inline proto/ibc/lightclients/beefy/v1/beefy.proto]: Nahhh this isn't right, clientstate already tracks this

File: modules/light-clients/11-beefy/types/update.go [hunk 2]
```diff
@@ -208,7 +208,7 @@ func (cs *ClientState) parachainHeadersToMMRProof(beefyHeader *Header) (*mmr.Pro
 		}
 
 		// not a fan of this but its golang
-		var parachainHeads = [32]byte{}
+		var parachainHeads SizedByte32
```
Comment (seunlanlege) [inline modules/light-clients/11-beefy/types/update.go:211]: And scale encoding/decoding works on these types?

File: proto/ibc/lightclients/beefy/v1/beefy.proto (hunk 1 above)
Comment (oshorefueled) [reply proto/ibc/lightclients/beefy/v1/beefy.proto]: Oops. Forgot about that, was supposed to be experimental.

File: modules/light-clients/11-beefy/types/update.go (hunk 2 above)
Comment (oshorefueled) [reply modules/light-clients/11-beefy/types/update.go:211]: Yes ser! Should we include tests for that?

Comment (seunlanlege) [reply modules/light-clients/11-beefy/types/update.go:211]: would prefer to see tests

**Comment by seunlanlege** [comment]
change the base to `seun-11-beefy`

**Pull request closed without merging**